Tribler/Core/CreditMining/CreditMiningManager.py

```diff
@@ -0,0 +1,269 @@
+import os
+import logging
+
+from glob import glob
+from binascii import unhexlify, hexlify
+from twisted.internet.task import LoopingCall
+from twisted.internet.defer import Deferred, DeferredList, succeed
+
+from Tribler.Core.CreditMining.CreditMiningSource import ChannelSource
+from Tribler.Core.CreditMining.CreditMiningPolicy import UploadPolicy, RandomPolicy
+from Tribler.Core.DownloadConfig import DefaultDownloadStartupConfig, DownloadStartupConfig
+from Tribler.Core.simpledefs import DLSTATUS_DOWNLOADING, DLSTATUS_STOPPED, DLSTATUS_SEEDING, DLSTATUS_STOPPED_ON_ERROR
+from Tribler.Core.TorrentDef import TorrentDefNoMetainfo
+from Tribler.dispersy.taskmanager import TaskManager
+
+
+class CreditMiningTorrent(object):
+    def __init__(self, infohash, name, download=None, state=None):
+        self.infohash = infohash
+        self.name = name
+        self.download = download
+        self.state = state
+        self.sources = set()
+        self.force_checked = False
+
+
+class CreditMiningSettings(object):
+    """
+    This class contains settings used by the credit mining manager
+    """
+    def __init__(self):
+        self.max_torrents_active = 8
+        self.max_torrents_listed = 100
+        # Note: be sure to set this interval to something that gives torrents a fair chance of
+        # discovering peers and uploading data
+        self.auto_manage_interval = 600
+        self.hops = 1
+        self.save_path = os.path.join(DefaultDownloadStartupConfig.getInstance().get_dest_dir(), 'credit_mining')
+
+
+class CreditMiningManager(TaskManager):
+    """
+    Class to manage all the credit mining activities
+    """
+
+    def __init__(self, session, settings=None):
+        super(CreditMiningManager, self).__init__()
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+        self.session = session
+        self.settings = settings or CreditMiningSettings()
+
+        self.sources = {}
+        self.torrents = {}
+        self.policies = []
+        self.num_checkpoints = 0
+
+        self.select_lc = self.register_task('select_torrents', LoopingCall(self.select_torrents))
+        self.session_ready = Deferred()
+
+    def initialize(self, policies=None):
```

Comment:
Maybe you could merge this method with the `__init__` method, developers may forget to initialize.